presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -208,6 +212,47 @@ public void testLongStrideDictionary()
         testRoundTripNumeric(concat(ImmutableList.of(1), nCopies(9999, 123), ImmutableList.of(2), nCopies(9999, 123)), BigintRange.of(123, 123, true));
     }
 
+    @Test
+    public void testFloatSequence()
+            throws Exception
+    {
+        tester.testRoundTrip(REAL, floatSequence(0.0f, 0.1f, 30_000));
+    }
+
+    @Test
+    public void testFloatNaNInfinity()
+            throws Exception
+    {
+        tester.testRoundTrip(REAL, ImmutableList.of(1000.0f, -1.23f, Float.POSITIVE_INFINITY));
+        tester.testRoundTrip(REAL, ImmutableList.of(-1000.0f, Float.NEGATIVE_INFINITY, 1.23f));
+        tester.testRoundTrip(REAL, ImmutableList.of(0.0f, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY));
+
+        tester.testRoundTrip(REAL, ImmutableList.of(Float.NaN, -0.0f, 1.0f));
+        tester.testRoundTrip(REAL, ImmutableList.of(Float.NaN, -1.0f, Float.POSITIVE_INFINITY));
+        tester.testRoundTrip(REAL, ImmutableList.of(Float.NaN, Float.NEGATIVE_INFINITY, 1.0f));
+        tester.testRoundTrip(REAL, ImmutableList.of(Float.NaN, Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY));
+    }
+
+    @Test
+    public void testFloatWithFilters()
```

Comment:
Could you fold this into other tests methods similar to other types? `tester.testRoundTrip` does multiple runs, one of which is with no filter.